### DISCUSSION TITLE
a simple extension for a programming language that has tree-sitter definitions

### DISCUSSION THREAD

**Discussion by hros** [microsoft/vscode-discussions#1037] [Extension Development QnA] [answered]
What is the best (easiest?) way to add syntax highlighting to a custom programming language for which I have the parse definition in tree-sitter format?
1. Is there a built-in vs-code capability for this?
2. Is there an existing extension that can support this?
3. If I need to develop a new extension, Is there an existing project (or demo) that can be used as reference?

can this new language be supported in the web version of vs-code (vscode.dev)?

Lastly, what would it take to add hover-tooltips of function definitions (from the api's docstrings)

**Comment by tamuratak** [comment, marked as answer]
You can refer to https://github.com/microsoft/vscode-anycode, a tree-sitter-based language extension

**Reply by hros** [reply]
thanks @tamuratak 
this sounds like a perfect solution for highlighting a new programming language in vscode.dev
1. does it also offer the ability to show the function's docstrings and API (argument list and types)?
2. building anycode (with `node ./scripts/all-npm.js i`) currently fails - I'll open an issue. After the issue is fixed, can you point me to instructions of adding the built extension to vscode.dev?

**Comment by abrilurena** [comment]
Hi, @hros!  I just wanted to check in and see if you still need any help with this?

**Reply by hros** [reply]
Looking into [Monaco editor](https://microsoft.github.io/monaco-editor/)
